sdk/src/main/scala/com/horizen/account/api/rpc/service/EthService.scala

```diff
@@ -786,6 +794,44 @@ class EthService(
         // apply requested transaction with tracing enabled
         tagStateView.applyTransaction(requestedTx, previousTransactions.length, gasPool, blockContext)
 
+        // return the tracer result from the evm
+        if(blockContext.getEvmResult != null) {
+          blockContext.getEvmResult.tracerResult
+        } else Unit
+      }
+    }
+  }
+
+  @RpcMethod("debug_traceCall")
+  @RpcOptionalParameters(1)
+  def traceCall(params: TransactionArgs, tag: String, config: TraceOptions): Any = {
```

Comment:
The expected behavior of `debug_traceCall` is inconsistent with all the other functions accepting blocknumber/blockhash/tag, right?
Everwhere else there is `getSomethingByBlockNumber` which accepts block number or tag and `getSomethingByBlockHash` which only accepts a block hash. Here it is all in one function.

Reply:
According to the _geth_ documentation the block can be specified either by hash or by number as the second argument (https://geth.ethereum.org/docs/interacting-with-geth/rpc/ns-debug#debugtracecall). Moreover, the `geth` current implementation seems to manage both block number or block hash (https://github.com/ethereum/go-ethereum/blob/master/eth/tracers/api.go#L895). 
This led to our current implementation that need to some modifications to interact with methods that manage by default a block by its number or tag.